handlers/search.go

```diff
@@ -364,32 +321,34 @@ func SearchAhead(w http.ResponseWriter, r *http.Request) {
 			WHERE validatorrow <= $2 AND namerow <= 10
 			GROUP BY name
 			ORDER BY count DESC, name DESC`, "%"+search+"%", searchValidatorsResultLimit)
-		if err == nil {
-			for i := range res {
-				res[i].Name = string(utils.FormatValidatorName(res[i].Name))
-			}
+		if err != nil {
+			break
+		}
+		for i := range res {
+			res[i].Name = string(utils.FormatValidatorName(res[i].Name))
 		}
 		result = &res
 	case "ens":
-		data, err := GetEnsDomain(search)
-		if err == nil {
-			result = &data
+		var data *types.EnsDomainResponse
+		data, err = GetEnsDomain(search)
+		if err != nil {
```

Comment:
This now introduces error logs like `time="2023-07-13T09:35:12+02:00" level=error msg="error doing query for searchAhead" error="not an ens domain or address" module=handlers searchType=ens` when searching for something like "a".

Furthermore, searching for valid ens names (for example "uhahah.eth") that are not used though results in `level=error msg="error doing query for searchAhead" error="sql: no rows in result set" module=handlers searchType=ens`.

Most likely you should not use `err` here but do something like the `convertErr` here: 
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/76e74910164abad2e678b45b693ce2ddaa81b913/handlers/search.go#L83

Reply:
should be fixed, ptal